tests/ops/test_stats.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+import torch
+
+from pyro.ops.stats import (autocorrelation, autocovariance, effective_sample_size,
+                            gelman_rubin, split_gelman_rubin,
+                            _cummin, _fft_next_good_size)
+from tests.common import assert_equal
+
+
+def test_gelman_rubin():
+    # only need to test precision for small data
+    x = torch.empty(2, 10)
+    x[0] = torch.arange(10.)
+    x[1] = torch.arange(10.) + 1
+
+    r_hat = gelman_rubin(x, sample_dim=1, chain_dim=0)
+    assert_equal(r_hat.item(), 0.98, prec=0.01)
+
+    # test shape
+    y = torch.rand(3, 7, 10)
+    assert_equal(gelman_rubin(y).shape, torch.Size([10]))
+    assert_equal(gelman_rubin(y, sample_dim=-1, chain_dim=0).shape, torch.Size([7]))
+    assert_equal(gelman_rubin(y, sample_dim=1, chain_dim=2).shape, torch.Size([3]))
+
+
+def test_split_gelman_rubin():
+    x = torch.rand(4, 10, 5)
+    r_hat1 = gelman_rubin(x.reshape(4, 2, 5, 5).reshape(8, 5, 5), sample_dim=1, chain_dim=0)
+    r_hat2 = split_gelman_rubin(x, sample_dim=1, chain_dim=0)
+    assert_equal(r_hat1, r_hat2)
+
+
+def test_autocorrelation():
+    x = torch.arange(10.)
```

Comment:
I understand that you have already validated the output wrt Stan / arviz?

Reply:
Yes, all the tests are against arviz. For efficient_sample_size, the results are a little bit different with different random seed (to generate `input`): the difference are 54 (ours) vs 57 (theirs), 95 vs 97, 992 vs 996 (kind of that).